cmd/cmd.go

```diff
@@ -0,0 +1,98 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package cmd implements command line utility for HarbourBridge.
+package cmd
+
+import (
+	"flag"
+	"fmt"
+	"time"
+
+	"github.com/cloudspannerecosystem/harbourbridge/conversion"
+	"github.com/cloudspannerecosystem/harbourbridge/internal"
+)
+
+var (
+	badDataFile = "dropped.txt"
+	reportFile  = "report.txt"
+	schemaFile  = "schema.txt"
+	sessionFile = "session.json"
+
+	schemaSampleSize = int64(0)
+)
+
+func init() {
+	flag.Int64Var(&schemaSampleSize, "schema-sample-size", int64(100000), "schema-sample-size: the number of rows to use for inferring schema (only for DynamoDB)")
+}
+
+// CommandLine provides the core processing for HarbourBridge when run as a command-line tool.
+// It performs the following steps:
+// 1. Run schema conversion (if dataOnly is set to false)
+// 2. Create database (if schemaOnly is set to false)
+// 3. Run data conversion (if schemaOnly is set to false)
+// 4. Generate report (if schemaOnly is set to false)
```

Comment:
Suggest dropping "(if schemaOnly is set to false)" since we generate a report in either case (see the call to report at line 61).

Reply:
Good catch! Done @ https://github.com/cloudspannerecosystem/harbourbridge/pull/121/commits/96fafc8df69686a5aeac12ee5bec2c5656666d1e.